src/cloud-api-adaptor/docs/SecureComms.md

```diff
@@ -0,0 +1,125 @@
+- Kubernetes Phase: In this phase, an SSH channel is created to enable the runtime to communicate with the Kata Agent running in the Peer Pod. This phase uses the keys obtained during the Attestation Phase, ensuring that the communication is secured.
+
+Additional tunnels can be established via the SSH channel by configuring the Adaptor and Forwarder as described below.
+
+## Secure Comms Architecture
+`Secure Comms` secures the communication between  the cluster Worker Nodes and Peer Pods. The Worker Nodes are assumed to be protected by a firewall and can be kept unreachable from outside of the cluster of Worker Nodes. To communicate with Peer Pods, an SSH channel is created from the Worker Node allocated to run the Peer Pod and the Peer Pod VM. This SSH channel can then be used for all communication between the Peer Pod and the cluster and any back end services running at the cluster or elsewhere. As shown in the diagram, once the SSH channel is created, clients at the Peer Pod VM side such as `KBC`, `Attestation Agent`, etc. may connect to services offered by the cluster side such as `KBS`, `Attestation Services`, etc. At the same time, clients located at the cluster side such as the `Runtime shim`, may connect to services offered by the Peer Pod VM side such as the `Kata Agent`.
+
+
+<p align="center">
+  <img width="500"  src="SecureComms.png">
+</p>
+
+SecureComms uses the following sequence of steps:
+- Worker Node creates keys for the Peer Pod and updates Trustee
+- Worker Node creates the Peer Pod VM
+- Worker Node establishes an "Attestation Phase" SSH channel with the Peer Pod VM
+- Peer Pod VM attests and obtain keys from Trustee. The Peer Pod then signal to the Worker Node that the "Attestation Phase" has ended and terminates the "Attestation Phase" SSH channel.
+- Worker Node establishes an "Kubernetes Phase" SSH channel with the Peer Pod VM  - Both sides verify the identity of the other side using the keys delivered via Trustee.
+
+Once the "Kubernetes Phase" SSH channel is established, Secure Comms connects the `Runtime shim` to the `Kata Agent` and may also connect other services required by the Peer Pod VM or by containers running inside the Peer Pod.
+
+See [Secure Comms Architecture Slides](./SecureComms.pdf) for more details.
+
+## Setup
+
+### Deploy CAA
+Use any of the option for installing CAA depending on the cloud driver used.
+
+### Deploy KBS-Operator
+Deploy KBS-Operator by following instructions at [KBS Operator Getting Started](https://github.com/confidential-containers/kbs-operator?tab=readme-ov-file#getting-started).
```

Comment:
I'm not sure if it's worth mentioning that users should uncomment the secrets generator since https://github.com/confidential-containers/kbs-operator/commit/40ed7645149bc10193e882deb66b0d293ba41cf6?